### ISSUE TITLE
多标签分类，使用头条1070类数据，loss下降很慢，验证集准确率也很低

### ISSUE THREAD

**Issue by AnddyWang** [yongzhuo/Keras-TextClassification#32]
各位有碰到过么，该怎么解决下呢

1）用categorical_accuracy来训练，这个准确率很低，有方案提升么？
loss: 5.6934 - categorical_accuracy: 0.3676 - val_loss: 11.7893 - val_categorical_accuracy: 0.2659

2）用top_k_categorical_accuracy来训练
loss: 7.0134 - top_k_categorical_accuracy: 0.8663 - val_loss: 9.7571 - val_top_k_categorical_accuracy: 0.7353

**Comment by yongzhuo** [comment]
你用binary_crossentropy作loss, accuracy作mertics试试呢

**Comment by AnddyWang** [comment]
@yongzhuo 'binary_crossentropy'应该不能用于多分类任务，这样得到的Accuary没有意义吧
刚开始训练loss就很低，loss: 0.0063 - acc: 0.9979
1)层次多标签分类(NHC)：是训多个二分类么？类别数有点多啊
2)可以大概说下seq2seq类方法构造标签这个实现模式么

**Comment by yongzhuo** [comment]
1. 这里项目的多标签分类按照一般的方式，转化为one-hot标签，比如说标签"news_fashion/dressing,news_entertainment/vulgar,news_fashion"， 转化为[1,1,1, 0, 0, 0, 0......]，所以可以用binary_crossentropy；
2. 不是训练是训多个二分类，NHC你可以参考腾讯开源的[NeuralNLP-NeuralClassifier](https://github.com/Tencent/NeuralNLP-NeuralClassifier)，它实现了一个NHC算法，大致可以理解为每一个层次的标签按照一定权重计算loss。
3. seq2seq类和强化学习合在一起用，主要是解决几个标签间依赖问题，你可以看看这个项目[https://github.com/lancopku/SGM](https://github.com/lancopku/SGM)

**Comment by AnddyWang** [comment]
@yongzhuo 
1）如果用了binary_crossentropy，直接用accuracy出来的准确率是不是不对呢，发现loss基本一直很低，变的很少？这个只能自己另外单独统计下？
2）3）好的，我看下，多谢了。
